spot_ros2_control/examples/src/wiggle_arm.cpp

```diff
@@ -0,0 +1,150 @@
+      throw std::logic_error("Wiggle offsets must be the same size as joints_to_wiggle!");
+    }
+
+    joint_states_sub_ = create_subscription<sensor_msgs::msg::JointState>(
+        "joint_states", 10, std::bind(&WiggleArm::joint_states_callback, this, std::placeholders::_1));
+    command_pub_ = create_publisher<std_msgs::msg::Float64MultiArray>("/forward_position_controller/commands", 10);
+    const auto timer_period =
+        std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::duration<double>(1. / command_rate));
+    timer_ = create_wall_timer(timer_period, std::bind(&WiggleArm::timer_callback, this));
+  }
+
+ private:
+  // stores joint angles and desired offsets
+  std::vector<double> nominal_joint_angles_;
+  std::vector<int64_t> joints_to_wiggle_;
+  std::vector<double> wiggle_up_offsets_;
+  std::vector<double> wiggle_down_offsets_;
+  size_t njoints_to_wiggle_;
+  // Command to send to the robot
+  std_msgs::msg::Float64MultiArray command_;
+  // Parameters
+  WiggleState wiggle_state_;
+  int points_per_motion_;
+  bool initialized_;
+  int count_;
+  // Timer, publishers, subscribers
+  rclcpp::TimerBase::SharedPtr timer_;
+  rclcpp::Subscription<sensor_msgs::msg::JointState>::SharedPtr joint_states_sub_;
+  rclcpp::Publisher<std_msgs::msg::Float64MultiArray>::SharedPtr command_pub_;
+
+  /// @brief Callback for receiving joint states messages used to store the nominal joint angles of the robot
+  /// @param msg ROS message containing joint states
+  void joint_states_callback(const sensor_msgs::msg::JointState& msg) {
+    if (!initialized_) {
+      RCLCPP_INFO_STREAM(get_logger(), "Received starting joint states");
+      nominal_joint_angles_ = msg.position;
+      command_.data = msg.position;
+      initialized_ = true;
+    }
+  }
+
+  /// @brief Rotates from WIGGLE_DOWN -> WIGGLE_MIDDLE -> WIGGLE_UP -> RESET -> WIGGLE_DOWN -> ...
+  /// WIGGLE_MIDDLE and RESET are moving towards the same set of nominal joint angles, just from different directions.
+  void state_transition() {
+    switch (wiggle_state_) {
+      case WiggleState::WIGGLE_DOWN:
+        wiggle_state_ = WiggleState::WIGGLE_MIDDLE;
+        break;
+      case WiggleState::WIGGLE_MIDDLE:
+        wiggle_state_ = WiggleState::WIGGLE_UP;
+        break;
+      case WiggleState::WIGGLE_UP:
+        wiggle_state_ = WiggleState::RESET;
+        break;
+      case WiggleState::RESET:
+        wiggle_state_ = WiggleState::WIGGLE_DOWN;
+        break;
+    }
+  }
+
+  /// @brief Fills in the command to send to the robot
+  /// @param offsets Vector of offsets to apply to joints_to_wiggle.
+  /// @param percentage Percentage of the motion (from 0-1) we are at. 0 corresponds to being at nominal_joint_angles_
+  /// and 1 corresponds to being at nominal_joint_angles_+offsets, anything in between is calculated as a linear
+  /// interpolation between the two.
+  void populate_command(std::vector<double>& offsets, double percentage) {
```

Comment:
@khughes-bdai nit: `offsets` can be made `const`:
```suggestion
  void populate_command(const std::vector<double>& offsets, double percentage) {
```